packages/sdk/schema/src/types/view.ts

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type MutableSchema } from '@dxos/echo-schema';
+import { getProperty } from '@dxos/effect';
+import { invariant } from '@dxos/invariant';
+
+import { schemaForType } from './field';
+import { type FieldType, type ViewType } from './types';
+
+export const addFieldToView = (
```

Comment:
I think this is right, you add a field to both view schema and type schema, but:

1. If view schema and type schema are in the same format, the code will look more similar.
2. I'm not sure about the current `schema.addFields` API as it doesn't allow for concurrent editing, perhaps we should just manipulate the JSONSchema stored in echo directly:

```ts
schema.jsonSchema.properties[newField] = { type: 'string', ... }
```